server/controllers/proxyController.ts

```diff
@@ -0,0 +1,91 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import httpProxy from 'http-proxy';
+import zlib from 'zlib';
+
+import { AirlockController, AirlockOptions } from '../main';
+import { AIRTABLE_API_BASE_URL } from '../utils/AirtableRoute';
+import cache from '../utils/cache';
+
+enum Encoding {
+  GZIP = 'gzip',
+  DEFLATE = 'deflate',
+}
+
+export default (options: AirlockOptions): AirlockController<{ web: {} }> => {
+  const { airtableApiKey } = options;
+  const proxy = httpProxy.createProxyServer({
+    changeOrigin: true,
+    ignorePath: false,
+    selfHandleResponse: true,
+  });
+
+  async function handlePayload(
+    buffer: Buffer,
+    encoding: string,
+  ): Promise<string> {
+    return new Promise((resolve, reject) => {
+      if (encoding === Encoding.GZIP) {
+        zlib.gunzip(buffer, function(err, decoded) {
+          if (err) return reject(err);
+          resolve(decoded.toString());
+        });
+      } else if (encoding === Encoding.DEFLATE) {
+        zlib.inflate(buffer, function(err, decoded) {
+          if (err) return reject(err);
+          resolve(decoded.toString());
+        });
+      } else {
+        resolve(buffer.toString());
+      }
+    });
+  }
+
+  proxy.on('proxyReq', function(proxyReq) {
+    proxyReq.setHeader('authorization', `Bearer ${airtableApiKey}`);
+  });
+  proxy.on('proxyRes', function(
+    proxyRes: IncomingMessage,
+    req: IncomingMessage,
+    res: ServerResponse,
+  ) {
+    var body = [];
+    proxyRes.on('data', function(chunk) {
+      body.push(chunk);
+    });
+    proxyRes.on('end', async function() {
+      const buffer = Buffer.concat(body);
+      const proxyPayload = await handlePayload(
+        buffer,
+        proxyRes.headers['content-encoding'],
+      );
+      if (req.method === 'GET' && proxyRes.statusCode === 200) {
+        cache.set(req, proxyPayload);
```

Comment:
Let's add a third parameter to cache.set, so that it takes in the proxyRes, so that the cache.set function can access the proxyRes headers.

with cache.set updated, this call should look like

`cache.set(req, proxyRes, proxyPayload);`